app/actions/remote/entry/gql_common.ts

```diff
@@ -98,6 +100,20 @@ export async function deferredAppEntryGraphQLActions(
     updateCanJoinTeams(serverUrl);
     updateAllUsersSince(serverUrl, since);
 
+    // defer sidebar GM profiles
+    setTimeout(async () => {
+        const gmIds = chData?.channels?.reduce<string[]>((acc, v) => {
+            if (v?.type === General.GM_CHANNEL) {
+                acc.push(v.id);
+            }
+            return acc;
+        }, []);
+        const channelsToFetchProfiles = new Set<string>(gmIds);
```

Comment:
```suggestion
        const channelsToFetchProfiles = chData?.channels?.reduce<Set<string>>((acc, v) => {
            if (v?.type === General.GM_CHANNEL) {
                acc.add(v.id);
            }
            return acc;
        }, new Set<string>());
```

I think this is better, no?